pyirf/interpolators/tests/test_base_interpolators.py

```diff
@@ -0,0 +1,184 @@
+
+    def dummy_extrapolator(target_point):
+        return np.sum(target_point)
+
+    class DummyBaseInterpolator(BaseInterpolator):
+        def interpolate(self, target_point, **kwargs):
+            raise NotImplementedError
+
+    grid_points1D_good = np.array([1, 2, 3])
+    target1D_outofGrid = np.array([0.5])
+
+    grid_points2D_good = np.array([[1, 1], [2, 1], [1.5, 1.5]])
+    target2D_outofGrid = np.array([0.5, 0.5])
+
+    # 1D target out of grid but extrapolator given
+    interp1D = DummyBaseInterpolator(grid_points1D_good)
+    assert interp1D(target1D_outofGrid, extrapolator=dummy_extrapolator) == np.sum(
+        target1D_outofGrid
+    )
+
+    # 2D target out of grid but extrapolator given
+    interp2D = DummyBaseInterpolator(grid_points2D_good)
+    assert interp2D(target2D_outofGrid, extrapolator=dummy_extrapolator) == np.sum(
+        target2D_outofGrid
+    )
+
+
+def test_ParametrizedInterpolator():
+    """Test ParametrizedInterpolator initialization and sanity checks"""
+    from pyirf.interpolators.base_interpolators import ParametrizedInterpolator
+
+    grid_points = np.array([1, 2, 3])
+    params_good = np.array([[1], [2], [3]])
+    params_shape_missmatch = np.array([[1], [2]])
+
+    with pytest.raises(TypeError):  # Abstract class, cannot be instantiated
+        ParametrizedInterpolator(grid_points, params_good)
+
+    class DummyParametrizedInterpolator(ParametrizedInterpolator):
+        def interpolate(self, target_point, **kwargs):
+            raise NotImplementedError
+
+    with pytest.raises(TypeError):  # parameters not a np.ndarray
+        DummyParametrizedInterpolator(grid_points, params_good.tolist())
+
+    with pytest.raises(ValueError):  # Fewer parameters then grid_points
+        DummyParametrizedInterpolator(grid_points, params_shape_missmatch)
+
+    interp = DummyParametrizedInterpolator(grid_points, params_good)
+
+    with pytest.raises(NotImplementedError):
+        # Everything ok but _interpolate not implemented
+        interp(np.array([1]))
+
+
+def test_BinnedInterpolator():
+    """Test BinnedInterpolator initialization and sanity checks"""
+    from pyirf.interpolators.base_interpolators import BinnedInterpolator
+
+    grid_points = np.array([1, 2, 3])
+    bin_edges = np.linspace(-1, 1, 11)
+    bin_content_good_shape = np.ones(shape=(len(grid_points), len(bin_edges) - 1))
+    bin_content_bad_nhist = np.ones(shape=(len(grid_points) - 1, len(bin_edges) - 1))
+    bin_content_bad_nbins = np.ones(shape=(len(grid_points), len(bin_edges)))
+
+    with pytest.raises(TypeError):  # Abstract class, cannot be instantiated
+        BinnedInterpolator(grid_points, bin_edges, bin_content_good_shape)
+
+    class DummyBinnedInterpolator(BinnedInterpolator):
+        def interpolate(self, target_point, **kwargs):
+            raise NotImplementedError
+
+    with pytest.raises(TypeError):  # bin_edges are not a np.ndarray
```

Comment:
Instead of having a comment with the reason, you can use `match="...."` to actually make sure you get the expected message